fs/fs_zenfs.cc

```diff
@@ -1304,7 +1317,124 @@ void ZenFS::GetZenFSSnapshot(ZenFSSnapshot& snapshot,
     zbd_->GetMetrics()->ReportSnapshot(snapshot);
   }
 
-  zbd_->LogGarbageInfo();
+  if (options.log_garbage_) {
+    zbd_->LogGarbageInfo();
+  }
+}
+
+IOStatus ZenFS::MigrateExtents(
+    const std::vector<ZoneExtentSnapshot*>& extents) {
+  IOStatus s;
+  // Group extents by their filename
+  std::map<std::string, std::vector<ZoneExtentSnapshot*>> file_extents;
+  for (auto* ext : extents) {
+    std::string fname = ext->filename;
+    // We only migrate SST file extents
+    if (ends_with(fname, ".sst")) {
+      file_extents[fname].emplace_back(ext);
+    }
+  }
+
+  for (const auto& it : file_extents) {
+    s = MigrateFileExtents(it.first, it.second);
+    if (!s.ok()) break;
+  }
```

Comment:
fixed

Reply:
Hmm, as empty zones are not allocated for migration in TakeMigrateZone, i don't see how ResetUnusedIOZones would help there.

To clarify my previous comment: When we have freed up space, we need to reset the zone(s) using ResetUnusedIOZones, otherwise the space won't be available to the writing threads. My suggestion is to do it here, in MigrateExtents before returning in stead.. (return ResetUnusedIOZones())